cie_infra/models/infra_instance_database.py

```diff
@@ -0,0 +1,89 @@
+from odoo import models
+from odoo import fields
+from odoo import api
+from odoo import _
+import requests
+import json
+from werkzeug.exceptions import BadRequest, InternalServerError, NotFound
+from odoo.exceptions import AccessDenied, Warning as UserError
+
+class Database(models.Model):
+    _name = 'infra.instance.database'
+    _description = 'database'
+    name=fields.Char('name', required=True)
+    instance_id=fields.Many2one('infra.instance', string='instance')
+    installed_module_ids=fields.One2many(
+        comodel_name='infra.instance.module.info',
+        inverse_name='database_id',
+        string='installed_module')
+
+    @api.multi
+    def http_get(self, url, params, headers):
+        self.ensure_one()
+        if url.startswith("/"):
+            url = self.url + url
+        return requests.get(url,params,headers)
+
+    def _process_response(self, response):
+        if response.status_code == 200:
+            content = response.content.decode("utf-8")
+            return json.loads(content)
+        elif response.status_code == 400:
+            content = response.content.decode("utf-8")
+            raise BadRequest("%s" % content)
+        elif response.status_code == 403:
+            raise AccessDenied(
+                _("You are not allowed to access this resource")
+            )
+        elif response.status_code == 404:
+            raise NotFound(
+                _("Resource not found %s on server" % response.status_code)
+            )
+        else:  # 500 et al.
+            content = response.content.decode("utf-8")
+            raise InternalServerError(_("%s" % content))
+
+    @api.multi
+    def http_get_content(self, url, params=None, headers=None):
+        self.ensure_one()
+        response = self.http_get(url, params=params, headers=headers)
+        return self._process_response(response)
+
+
+    def cron_get_module(self):
+        """
+        thanks to http request, the cron will recover daily
+        the different database based on a odoo's instance
+        """
+
+        databases = self.env['infra.instance.database'].search([])
+        for database in databases:
+            route = "/server-info/databases/"+database.name+"/modules"
+            modules = database.http_get_content(route)
+            for module in modules:
+                if not self.is_module_already_installed(module):
+                    new_module = self.env['infra.instance.module'].create({
+                        "name": module["name"],
+                    })
+                    module_info=self.env["infra.instance.module.info"].create({
+                        "installed_version": module["published_version"],
+                    })
+                    new_module.write({
+                        "module_info_ids": module_info.id,
```

Comment:
Attention, ici `module_info_ids` contient plusieurs `module.info`. Donc comme tu fais là, tu supprimes tous les modules_info, et tu en lie uniquement un.

Aussi, tu dois passer l'objet directement et pas juste l'id de l'objet. :)

Reply:
d'apres la documentation odoo, je trouve cette options pour la fonction write : 
(0, 0, values)  adds a new record created from the provided value dict.
''' updated_module=[0,0,{"module_info_ids" : module_info}]
                    new_module.write(
                        updated_module
                    )'''

est ce la solution ou je dois override la function write pour lui donner le bon comportement?